narwhals/_spark_like/dataframe.py

```diff
@@ -274,6 +275,14 @@ def select(
     ) -> Self:
         new_columns = evaluate_exprs(self, *exprs)
         new_columns_list = [col.alias(col_name) for (col_name, col) in new_columns]
+        if not self._implementation.is_sqlframe():
```

Comment:
`sqlframe` doesn't seem to raise any error at `select` time. I guess it gets triggered during collection

@MarcoGorelli do you have any suggestions about what to do in this case?

Reply:
I think I have an idea. I can test it like we do for polars lazy frames 🤔